src/components/sliding-panel.tsx

```diff
@@ -0,0 +1,121 @@
+import { FC, useRef, useEffect, ReactNode } from 'react';
+import { createPortal } from 'react-dom';
+import cn from 'classnames';
+
+import { Button, CloseIcon } from './index';
+
+import '../styles/components/sliding-panel.scss';
+
+type LRBelowHeader = {
+  /**
+   * Where the sliding panel should appear
+   */
+  position: 'left' | 'right';
+  /**
+   * Horizontal position of the arrow. If the panel appears below the page header.
+   * Also works as a flag to display the arrow and display below the header
+   */
+  arrowX?: number;
+};
+
+type TBSlidingPanel = {
+  /**
+   * Where the sliding panel should appear
+   */
+  position: 'top' | 'bottom';
+  arrowX?: never;
+};
+
+type SlidingPanelProps = {
+  /**
+   * What happens when close is triggered. Responsability of the user of the compoent
+   */
+  onClose: (arg: void) => void;
+  /**
+   * Size of the panel once opened
+   */
+  size?: 'small' | 'medium' | 'large' | 'full-screen';
+  /**
+   * Title of the panel
+   */
+  title?: ReactNode;
+  /**
+   * Add a close button in the panel header
+   */
+  withCloseButton?: boolean;
+  /**
+   * Should the panel be scrollable if the content is taller than the panel?
+   */
+  yScrollable?: boolean;
+  className?: string;
+} & (LRBelowHeader | TBSlidingPanel);
+
+const SlidingPanel: FC<SlidingPanelProps> = ({
+  children,
+  onClose,
+  position,
+  size = 'medium',
+  title,
+  withCloseButton = false,
+  className,
+  yScrollable = false,
+  arrowX,
+}) => {
+  const node = useRef<HTMLDivElement>(null);
+
+  const onCloseRef = useRef(onClose);
+  onCloseRef.current = onClose;
+
+  useEffect(() => {
+    const handleClickOutside = (e: MouseEvent) => {
+      if (node?.current && !node.current.contains(e.target as Node)) {
+        e.stopPropagation();
+        e.preventDefault();
+        onCloseRef.current();
+      }
+    };
+
+    document.addEventListener('click', handleClickOutside, true);
+    return () => {
+      document.removeEventListener('click', handleClickOutside, true);
+    };
+  }, []);
+
+  return createPortal(
+    <div
+      data-testid="sliding-panel"
+      className={cn(
+        'sliding-panel',
+        `sliding-panel--${position}`,
+        `sliding-panel--${position}--${size}`,
+        Number.isFinite(arrowX) && `sliding-panel--${position}--below-header`,
+        className
+      )}
+      style={{ overflowY: yScrollable ? 'auto' : 'initial' }}
```

Comment:
When looking at the [sliding panel with arrow story](http://localhost:6006/?path=/story/layout-sliding-panel--sliding-panel-with-arrow) if `arrowX > 0` and `yScrollable` is toggled the panel's corner radius changes and the arrow disappears. What is the desired behaviour in this situation?

Reply:
In the end with the added header I don't really see any point in having `yScrollable` so removed it and made the content div scrollable on overflow. Also had to add flex to the parent so the overflow would work.